packages/shared/lib/core/nfts/actions/loadNftsForActiveProfile.ts

```diff
@@ -23,16 +23,31 @@ async function loadNftsForAccount(account: IAccountState): Promise<void> {
                 account.depositAddress,
                 outputData.output
             )
-            const nft = buildNftFromNftOutput(outputData.output, outputData.outputId, isSpendable, timeLockTime)
+            const nft = buildNftFromNftOutput(
+                outputData.output,
+                outputData.outputId,
+                isSpendable,
+                Number(outputData.output.amount),
+                timeLockTime
+            )
             accountNfts.push(nft)
         }
     }
+
     const allOutputs = await account.outputs()
-    for (const outputData of allOutputs) {
+    const sortedNftOutputs = allOutputs
+        .filter((output) => output.output.type === OUTPUT_TYPE_NFT)
+        .sort((a, b) => b.metadata.milestoneTimestampBooked - a.metadata.milestoneTimestampBooked)
+    for (const outputData of sortedNftOutputs) {
         if (outputData.output.type === OUTPUT_TYPE_NFT) {
             const nftId = getNftId(outputData.output.nftId, outputData.outputId)
             if (!accountNfts.some((nft) => nft.id === nftId)) {
-                const nft = buildNftFromNftOutput(outputData.output, outputData.outputId, false)
+                const nft = buildNftFromNftOutput(
+                    outputData.output,
+                    outputData.outputId,
+                    false,
+                    Number(outputData.output.amount)
```

Comment:
Why does `buildNftFromNftOutput` take in so many arguments that are on the same object?